src/Dialogs/RemoveFilesDialog.vala

```diff
@@ -26,94 +26,61 @@
  * Authored by: Scott Ringwelski <sgringwe@mtu.edu>
  */
 
-public class Noise.RemoveFilesDialog : Gtk.Dialog {
+public class Noise.RemoveFilesDialog : Granite.MessageDialog {
     public signal void remove_media (bool response);
 
-    private Gtk.Box content;
-
-    private Gtk.Button remove_button;
-    private Gtk.Button trash_button;
-    private Gtk.Button cancel_button;
-
     public RemoveFilesDialog (Gee.Collection<Media> to_remove) {
-        this.set_modal (true);
-        this.set_transient_for (App.main_window);
-        this.destroy_with_parent = true;
-        resizable = false;
-        deletable = false;
-
-        content = new Gtk.Box (Gtk.Orientation.VERTICAL, 10);
-
-        Gtk.Box padding = get_content_area () as Gtk.Box;
-        padding.set_orientation (Gtk.Orientation.HORIZONTAL);
-
-        // initialize controls
-        Gtk.Image warning = new Gtk.Image.from_icon_name ("dialog-warning", Gtk.IconSize.DIALOG);
-        Gtk.Label title = new Gtk.Label ("");
-        Gtk.Label info = new Gtk.Label ("");
-        trash_button = new Gtk.Button.with_label (_("Move to Trash"));
-        remove_button = new Gtk.Button.with_label (_("Remove from Library"));
-        cancel_button = new Gtk.Button.with_label (_("Cancel"));
-
-        bool multiple_media = to_remove.size > 1;
-
-        // set title text
-        title.halign = Gtk.Align.START;
-        string title_text = "";
-
-        if (multiple_media) {
-            title_text = _("Remove %d Songs From Library?").printf (to_remove.size);
+        Object (
+            destroy_with_parent: true,
+            image_icon: new ThemedIcon ("dialog-warning"),
+            modal: true,
+            primary_text: "",
+            secondary_text: "",
+            transient_for: App.main_window
+        );
+
+        primary_label.max_width_chars = 65;
+        secondary_label.max_width_chars = 65;
+
+        var to_remove_size = to_remove.size;
+
+        if (to_remove_size > 1) {
+            primary_text = ngettext (
+                "Remove %d Songs From Library?",
+                "Remove %d Songs From Library?",
```

Comment:
Shouldn't the first one be `Remove One Song From Library?`?

Reply:
Or at least `Remove %d Song from Library?`

Reply:
The singular one shouldn't actually get used since it's the totally different string above in this case, but yeah you're right